File: 백준/Silver/10799. 쇠막대기/쇠막대기.py
import sys
input = sys.stdin.readline

arr = input().rstrip()
stack = []
cnt = 0

for i in range(len(arr)):
    if arr[i] == '(':
        stack.append(i)
    else:
        if arr[i-1] == '(':
            stack.pop()
            cnt += len(stack)
        else:
            stack.pop()
            cnt += 1
print(cnt)

File: 백준/Silver/10799. 쇠막대기/쇠막대기.java
import java.util.Scanner;
import java.util.Stack;

public class Main {

	public static void main(String[] args) {
		Scanner sc = new Scanner(System.in);
		char[] sticks = sc.next().toCharArray();
		Stack<Character> stack = new Stack<Character>();
		int cnt = 0;
		
		for (int i=0; i<sticks.length; i++) {
			if (sticks[i] == '(') {
				stack.push(sticks[i]);
			}
			else {
				stack.pop();
				if (sticks[i-1] == '(') {
					cnt += stack.size();
				}
				else {
					cnt += 1;
				}
			}
		}
        System.out.println(cnt);
		sc.close();
	}
}
